query/join.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"github.com/square/metrics/api"
+)
+
+type JoinRow struct {
```

Comment:
i think `JoinRow`, `Join`, and `JoinResult` can all start with lowercase, because I don't think it needs to be used outside the package.

(go uses the capitalization of the first character to determine whether a given symbol is exported or not).